src/request.rs

```diff
@@ -91,21 +91,21 @@ pub(crate) async fn request<Input: Serialize + std::fmt::Debug, Output: 'static
         Ok(response) => Response::from(response),
         Err(e) => {
             error!("Network error: {:?}", e);
-            return Err(Error::UnreachableServer)
-        },
+            return Err(Error::UnreachableServer);
+        }
     };
     let status = response.status() as u16;
     let text = match response.text() {
         Ok(text) => match JsFuture::from(text).await {
             Ok(text) => text,
             Err(e) => {
                 error!("Invalid response: {:?}", e);
-                return Err(Error::Unknown("Invalid response".to_string()));
+                return Err(Error::HttpError("Invalid response".to_string()));
```

Comment:
It may be the right time to remove these ugly things. What do you think?

Reply:
Would it be reasonable to just store the `web-sys` error type?  I'm not sure if that would break anything - if that doesn't work, we could try converting it to a string and using the current system but with a better error message.

Reply:
The web-sys error type is awful and useless. It would be better to create an enum `WasmHttpError` or something like this.

Reply:
I don't know enough about the potential http errors to be able to create a custom error enum for that.  For now we should either use the current system, pass the `web-sys` error on to the user, or store a stringified version of the web-sys error.  Implementing a custom error type for web errors seems like a bad idea, as it will likely miss some or many cases.